### PR TITLE
Boyscout/mem tracking

### PR DESCRIPTION
**Pull request by RisingLiberty**

### CODE REVIEW

File: source/1_engine/rex_engine/include/rex_engine/diagnostics/stacktrace.h [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+#ifdef REX_PLATFORM_WINDOWS
+  #include "rex_engine/diagnostics/win/win_stacktrace.h"
```
Comment (Dyronix) [inline source/1_engine/rex_engine/include/rex_engine/diagnostics/stacktrace.h:4]: Do we want these conditional compilation statements or do we exclude the file in general from the build?

File: source/1_engine/rex_engine/include/rex_engine/diagnostics/win/win_stacktrace.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#pragma once
```
Comment (Dyronix) [inline source/1_engine/rex_engine/include/rex_engine/diagnostics/win/win_stacktrace.h:1]: Some comment as above

File: source/1_engine/rex_engine/include/rex_engine/memory/memory_header.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#include "rex_std_extra/memory/memory_size.h"
+
+namespace rex
+{
+  class MemoryHeader
+  {
+  public:
+    MemoryHeader(MemoryTag tag, void* ptr, rsl::memory_size size, rsl::thread::id threadId, card32 frameIdx, const win::CallStack& callstack);
+
+    const win::CallStack& callstack() const;
```
Comment (Dyronix) [inline source/1_engine/rex_engine/include/rex_engine/memory/memory_header.h:16]: Platform specific code should not be in the header

File: source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    // amount of page faults that have occurred
+    // Note: this doesn't differentiate between hard and soft page faults
+    // so don't assume that a high page fault count is bad
+    card32 num_page_faults{};
+  };
+} // namespace rex
+
+#ifdef REX_PLATFORM_WINDOWS
+  #include "rex_engine/memory/win/win_mem_stats.h"
```
Comment (Dyronix) [inline source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h:36]: Do we want these conditional compilation statements or do we exclude the file in general from the build?

File: source/1_engine/rex_engine/include/rex_engine/memory/win/win_mem_stats.h [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#pragma once
```
Comment (Dyronix) [inline source/1_engine/rex_engine/include/rex_engine/memory/win/win_mem_stats.h:1]: Do we want these conditional compilation statements or do we exclude the file in general from the build?

File: source/1_engine/rex_engine/include/rex_engine/platform.h [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+namespace rex
+{
+#ifdef REX_PLATFORM_WINDOWS
```
Comment (Dyronix) [inline source/1_engine/rex_engine/include/rex_engine/platform.h:5]: Do we want these conditional compilation statements or do we exclude the file in general from the build?

**Review by Dyronix — CHANGES_REQUESTED** [review]

File: source/1_engine/rex_engine/include/rex_engine/diagnostics/stacktrace.h (hunk 1 above)
Comment (Dyronix) [reply source/1_engine/rex_engine/include/rex_engine/diagnostics/stacktrace.h:4]: Shouldn't there be a platform header somewhere?

Comment (RisingLiberty) [reply source/1_engine/rex_engine/include/rex_engine/diagnostics/stacktrace.h:4]: I created an issue for excluding platform specific code from build.
I don't know what you mean with platform header

File: source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h (hunk 4 above)
Comment (RisingLiberty) [reply source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h:36]: I don't know how you'd exclude this from build other than what I did above, it's a header file, that's never compiled standalone. you can only exclude source files from build

File: source/1_engine/rex_engine/include/rex_engine/memory/memory_header.h [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+#include "rex_engine/platform.h"
+#include "rex_std/thread.h"
+#include "rex_std_extra/memory/memory_size.h"
+
+namespace rex
+{
+  class MemoryHeader
+  {
+  public:
+    MemoryHeader(MemoryTag tag, void* ptr, rsl::memory_size size, rsl::thread::id threadId, card32 frameIdx, const win::CallStack& callstack);
```
Comment (Dyronix) [inline source/1_engine/rex_engine/include/rex_engine/memory/memory_header.h:14]: Place replace any platform namespaces such as "win::" with a generic namespace such as "platform::"

File: source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h (hunk 4 above)
Comment (Dyronix) [reply source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h:36]: I mean exclude it from the project, if it's not present we cannot build it.

**Review by Dyronix — CHANGES_REQUESTED** [review]

File: source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h (hunk 4 above)
Comment (RisingLiberty) [reply source/1_engine/rex_engine/include/rex_engine/memory/memory_stats.h:36]: that's not really true, we use ninja, not vs for building, ninja doesn't have such a thing as projects.
all files that are in the "win" folder are excluded if the target platform is not windows

**Pull request merged**